Add letsencrypt fields to domain (#23)
Scalingo/developers-documentation: _scalingo_api/domains.md

**Keys attributes**

{:.table}
| field                | type     | description                                                       |
| -------------------- | -------- | ----------------------------------------------------------------- |
| id                   | string   | unique ID of the domain                                           |
| name                 | string   | hostname your want to associate with the app                      |
| tlscert (read)       | string   | subject of the submitted certificate                              |
| tlscert (write)      | string   | content of the SSL certificate you want to use                    |
| tlskey (read)        | string   | private key type and length                                       |
| tlskey (write)       | string   | content of the private key used to generate the certificate       |
| ssl (read-only)      | bool     | flag if SSL with a custom certificate is enabled                  |
| validity (read-only) | datetime | once a certificate has been submitted, display the validity of it |
| canonical            | bool     | the domain is the canonical domain of this application            |

> | letsencrypt          | bool     | the domain is using a let's encrypt status                        |

Reviewer: I don't really understand what this boolean is? What does it mean for a domain to use a Let's Encrypt status?

> | letsencrypt          | bool     | the domain is using a let's encrypt status                        |

Reviewer: ```suggestion
| letsencrypt          | bool     | the domain is using a Let's Encrypt status                        |
```

> | letsencrypt_status   | string   | let's encrypt certificate generation status                       |

Reviewer: ```suggestion
| letsencrypt_status   | string   | Let's Encrypt certificate generation status                       |
```

| acme_dns_fqdn        | string   | ACME DNS-01 TXT entry FQDN                                        |
| acme_dns_value       | string   | ACME DNS-01 TXT entry value                                       |
| acme_dns_error       | string   | ACME DNS-01 error                                                 |

> The letsencrypt_status field can take different values depending on your certificate state:

Reviewer: ```suggestion
The `letsencrypt_status` field can take different values depending on your certificate state:
```

- `pending_dns`: Scalingo is waiting for your DNS configuration to be correct
- `new`: The certificate request has been sent to Let's Encrypt

> - `created`: The certificate has been created by let's encrypt and is available

Reviewer: ```suggestion
- `created`: The certificate has been created by Let's Encrypt and is available
```

> - `dns_required`: (wildcard only) we're waiting for DNS update

Reviewer: ```suggestion
- `dns_required`: (wildcard only) Scalingo is waiting for DNS configuration update
```
